### PR TITLE
Add SSO registration scopes from the shared config file to CredentialProfileOptions.

### PR DESCRIPTION
**Pull request by floralph**
Added CredentialProfileOptions.SsoRegistrationScopes property that is set by SharedCredentialsFile.TryGetProfile.

## Description
Like the other SSO-related fields (sso_region and sso_start_url in an [sso-session] section in the shared config file, this change makes the sso_registration_scopes that may also be defined in the [sso-session] available on CredentialProfileOptions instances.  

This is for reading purposes only as there is no way to save an [sso-session] section from the SDK other than manually creating the section via ProfileIniFile, so no changes have been made for writing this field.

## Motivation and Context
This change is to make sso_registration_scopes from [sso-session] sections in the the shared config file available in code for comparison and filtering.

## Testing
Created a console app to verify that the CredentialFileOptions.SsoRegistrationScopes property matched that of the [sso-session] defined in my shared config file.  Updated SharedConfigurationFile tests to include SsoRegistrationScopes property.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist
- [x] My code follows the code style of this project
- [x] My change requires a change to the documentation (should just be API reference generated from XML doc comments)
- [ ] I have updated the documentation accordingly
- [x] I have read the **README** document
- [x] I have added tests to cover my changes
- [ ] All new and existing tests passed (there appears to be some concurrency issues with unit tests even before I made my changes; the tests I changed passed and continue to pass after the changes)

## License
- [x] I confirm that this pull request can be released under the Apache 2 license

[issues]: https://github.com/aws/aws-sdk-net/issues
[license]: http://aws.amazon.com/apache2.0/
[cla]: http://en.wikipedia.org/wiki/Contributor_License_Agreement

### CODE REVIEW

File: sdk/src/Core/Amazon.Runtime/CredentialManagement/CredentialProfileOptions.cs [hunk 1]
```diff
@@ -121,6 +121,11 @@ public class CredentialProfileOptions
         /// </summary>
         public string SsoRegion { get; set; }
 
+        /// <summary>
+        /// The SSO scopes that are provided for authorization when using AWS SSO.
+        /// </summary>
+        public string SsoRegistrationScopes { get; set; }
```
Comment (normj) [inline sdk/src/Core/Amazon.Runtime/CredentialManagement/CredentialProfileOptions.cs:127]: Can you update the `ToString`, `Equals` and `HashCode` methods to take include this new property.

**Review by normj — CHANGES_REQUESTED** [review]

File: sdk/src/Core/Amazon.Runtime/CredentialManagement/CredentialProfileOptions.cs (hunk 1 above)
Comment (floralph) [reply sdk/src/Core/Amazon.Runtime/CredentialManagement/CredentialProfileOptions.cs:127]: Done

**Review by normj — APPROVED** [review]

File: sdk/src/Core/Amazon.Runtime/CredentialManagement/CredentialProfileOptions.cs (hunk 1 above)
Comment (ashovlin) [reply sdk/src/Core/Amazon.Runtime/CredentialManagement/CredentialProfileOptions.cs:127]: And can you update `IsEmpty` as well?

File: sdk/src/Core/Amazon.Runtime/CredentialManagement/NetSDKCredentialsFile.cs [hunk 2]
```diff
@@ -63,6 +63,7 @@ public class NetSDKCredentialsFile : ICredentialProfileStore
 
         private const string SsoAccountId = "sso_account_id";
         private const string SsoRegion = "sso_region";
+        private const string SsoRegistrationScopes = "sso_registration_scopes";
```
Comment (ashovlin) [inline sdk/src/Core/Amazon.Runtime/CredentialManagement/NetSDKCredentialsFile.cs:66]: Can you add this to `ReservedPropertyNames` alongside the others for SSO?

File: sdk/src/Core/Amazon.Runtime/CredentialManagement/CredentialProfileOptions.cs (hunk 1 above)
Comment (floralph) [reply sdk/src/Core/Amazon.Runtime/CredentialManagement/CredentialProfileOptions.cs:127]: Done

File: sdk/src/Core/Amazon.Runtime/CredentialManagement/NetSDKCredentialsFile.cs (hunk 2 above)
Comment (floralph) [reply sdk/src/Core/Amazon.Runtime/CredentialManagement/NetSDKCredentialsFile.cs:66]: Done

**Review by ashovlin — APPROVED** [review]

**Comment by ashovlin** [comment]
This has been released in https://www.nuget.org/packages/AWSSDK.Core/3.7.203 via https://github.com/aws/aws-sdk-net/commit/68aef4e681fda621f54058d1f2240306cbc72352, https://github.com/aws/aws-sdk-net/commit/09346de46a68dde4bf648d6a5eb2f55b0f5b8498, and https://github.com/aws/aws-sdk-net/commit/0de21ad3e27b23f35a7411582ced81640798a2cd. The commit hashes changed when we rebased during the release, so GitHub didn't auto-close.

Thanks for the contribution!

**Pull request closed without merging**